tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/PostServiceImpl.java

```diff
@@ -664,7 +666,16 @@ public PostDTO findOne(Long id) {
   @Override
   public void delete(Long id) {
     log.debug("Request to delete Post : {}", id);
-    postRepository.deleteById(id);
+    Optional<Post> optionalPost = postRepository.findById(id);
+    if (optionalPost.isPresent()) {
+      Set<Placement> attachedPlacements = optionalPost.get().getPlacementHistory();
+      if (attachedPlacements.isEmpty()) {
+        postRepository.deleteById(id);
+        // Related PostFunding records will be cascade deleted
```

Comment:
It has also been tested in the local.

Reply:
Comment seems unnecessary.  The same is true of all attached entities marked with the appropriate annotations.